lib/ansible/modules/cloud/amazon/iam_cert.py

```diff
@@ -269,9 +289,9 @@ def main():
     new_path = module.params.get('new_path')
     dup_ok = module.params.get('dup_ok')
     if state == 'present' and not new_name and not new_path:
-        cert = module.params.get('cert')
-        key = module.params.get('key')
-        cert_chain = module.params.get('cert_chain')
+        cert, key, cert_chain = load_data(cert=module.params.get('cert'),
```

Comment:
I've also double checked what happens in the scenario you mentioned, and below is the result I receive.  So I actually this this method is fine, and there is no need to change it

Apart from ensuring the params are not None before checking if they are an existing file.

```
BotoServerError: 400 Bad Request\n<ErrorResponse xmlns=\"https://iam.amazonaws.com/doc/2010-05-08/\">
<Error>
    <Type>Sender</Type>
    <Code>MalformedCertificate</Code>
    <Message>Unable to parse certificate. Please ensure the certificate is in PEM format.</Message>
  </Error>
```